Silicon/NXP/iMXPlatformPkg/Drivers/SmbiosConfigDxe/SmbiosConfigDxe.c

```diff
@@ -0,0 +1,705 @@
+*  http://opensource.org/licenses/bsd-license.php
+*
+*  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+*
+**/
+
+#include "SmbiosConfigDxe.h"
+
+#define MAX_SMBIOS_KEY_REV_1 5
+CHAR8* mSmbiosKeyRev1[MAX_SMBIOS_KEY_REV_1] = {"System Manufacturer",
+                                               "System Product Name",
+                                               "System SKU",
+                                               "System Family",
+                                               "Baseboard Product"};
+
+STATIC CONST CHAR16 mSmbiosOverridePresent[] = L"SmbiosOverridePresent";
+
+STATIC SMBIOS_OVERRIDE_NODE *mSmbiosOverrideListHead = NULL;
+
+/**
+
+  Finds and retrieves the next Key/Value pair from the buffer starting at the
+  given offset.
+
+  This routine scans through the provided buffer starting at the given offset.
+  It finds the first '"' character and records its location as the start of the Key.
+  Then it continues scanning to find the next '"' instance. The length denotes the KeyLength.
+
+  We repeat these steps to find the Value position and length.
+
+  At any point if a failure is detected, the algorithm bails with an error.
+
+  If we successfully find both Key and Value positions and lengths, then allocate/copy
+  these to their respective buffers and return to the caller.
+
+**/
+EFI_STATUS
+EFIAPI
+GetNextKeyValuePair (
+  CHAR8 *Buffer,
+  UINTN *Offset,
+  UINTN BufferSize,
+  CHAR8 **Key,
+  CHAR8 **Value
+  )
+{
+  EFI_STATUS Status;
+  UINTN CurrentOffset;
+  UINTN KeyLength;
+  UINTN ValueLength;
+  JSON_PARSER_STATE State;
+  CHAR8 *KeyStart;
+  CHAR8 *ValueStart;
+  CHAR8 *TempKey;
+  CHAR8 *TempValue;
+
+  CurrentOffset = *Offset;
+  KeyLength = 0;
+  ValueLength = 0;
+  Status = EFI_NOT_FOUND;
+  State = StateFindKey;
+  KeyStart = NULL;
+  ValueStart = NULL;
+  TempKey = NULL;
+  TempValue = NULL;
+
+  while (CurrentOffset < BufferSize) {
+    if (State == StateValueDone) {
+      Status = EFI_SUCCESS;
+      break;
+    }
+
+    switch (State) {
+      case StateFindKey:
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateReadKey;
+        }
+      break;
+      case StateReadKey:
+        if (KeyStart == NULL) {
+          KeyStart = &Buffer[CurrentOffset];
+        }
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateKeyDone;
+        } else {
+          KeyLength++;
+        }
+      break;
+      case StateKeyDone:
+        if (Buffer[CurrentOffset] == ':') {
+          State = StateFindValue;
+        }
+      break;
+      case StateFindValue:
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateReadValue;
+        }
+      break;
+      case StateReadValue:
+        if (ValueStart == NULL) {
+          ValueStart = &Buffer[CurrentOffset];
+        }
+        if (Buffer[CurrentOffset] == '"') {
```

Comment:
Values can't contain quote characters.  No escaping mechanism.

Reply:
Discussed this with Daniel a bit. Seems like the best move is to simplify to an INI parser with just simple key=value\n setup and remove this JSON parsing attempt. The INI parser will treat the Value as a literal so no escape characters needed.
Sample:
```
Signature=SmbiosOverrideConfigurationTable
Revision=1
System Manufacturer=Contoso
System Family=Contoso Family
```